quickwit/quickwit-janitor/src/garbage_collection.rs

```diff
@@ -21,25 +21,29 @@ use std::path::Path;
 use std::sync::Arc;
 use std::time::Duration;
 
-use futures::{Future, StreamExt};
+use futures::Future;
 use quickwit_actors::ActorContext;
 use quickwit_metastore::{Metastore, MetastoreError, SplitMetadata, SplitState};
-use quickwit_storage::{Storage, StorageError};
+use quickwit_storage::{BulkDeleteError, Storage};
 use serde::Serialize;
 use thiserror::Error;
 use time::OffsetDateTime;
+use tokio::sync::{mpsc, Semaphore};
 use tracing::error;
 
 use crate::actors::GarbageCollector;
 
 const MAX_CONCURRENT_STORAGE_REQUESTS: usize = if cfg!(test) { 2 } else { 10 };
 
+/// The number of split delete requests to combine and issue as a single bulk request.
+const BULK_DELETE_CHUNK_SIZE: usize = 10;
```

Comment:
Currently, the chunk size is set to `10` which is pretty arbitrary, in S3 the maximum number per request is `1000` but this may be different for each blob store and is also probably overkill. I would say some number between 10 - 100 is sufficient but it will want some load testing to really know how much of a difference it's actively making.

Another option is we make this user configurable, but that adds an additional set of hoops for us to jump through and is probably not critical enough to be worth exposing to the user.